src/openapi/streaming/connection/transport/websocket-transport.js

```diff
@@ -25,8 +25,50 @@ const DEFAULT_RECONNECT_LIMIT = 10;
 
 const NOOP = () => {};
 
+let maxCallstackSize;
+
 // -- Local methods section --
 
+function getMaxCallstackSize() {
+    if (maxCallstackSize) {
+        return maxCallstackSize;
+    }
+
+    let index = 0;
+    function recurse() {
+        index++;
+        recurse();
+    }
+
+    try {
+        recurse();
+    } catch (e) {
+        maxCallstackSize = index;
+    }
+
+    return maxCallstackSize;
+}
+
+function getJSONPayloadString(payloadBuffer) {
+    const chunkSize = getMaxCallstackSize();
+    const chunks = Math.ceil(payloadBuffer.length / chunkSize);
+    let payload = '';
+    let chunkIndex = 0;
+
+    while (chunkIndex < chunks) {

```

Comment:
we should have a test covering this - which will also be easier with a smaller chunk size

Reply:
sure
will add it